tests/test_write.py

```diff
@@ -126,6 +126,25 @@ class Writer:
     assert Writer.writerow.call_count == len(data)
 
 
+@patch("csv.writer")
+def test_write_dict(mock_save, tmpdir):
+    """Test the write_csv function."""
+    from csvy.writers import write_dict
+
+    class Writer:
+        writerow = MagicMock()
+
+    mock_save.return_value = Writer
+    filename = tmpdir / "some_file.csv"
+
+    data = {"a": [1, 2, 3, 4], "b": [1, 2, 3], "c": [1, 2, 3, 4, 5]}
+    expected_rows = max(len(v) for v in data.values()) + 1  # +1 for the column names
```

Comment:
Not important, but you could write this as:

```suggestion
    expected_rows = max(map(len, data.values())) + 1  # +1 for the column names
```